mp4parse/src/lib.rs

```diff
@@ -3031,16 +3038,16 @@ fn read_av1c<T: Read>(src: &mut BMFFBox<T>) -> Result<AV1ConfigBox> {
     let chroma_subsampling_x = (flags_byte & 0x08) >> 3;
     let chroma_subsampling_y = (flags_byte & 0x04) >> 2;
     let chroma_sample_position = flags_byte & 0x03;
-    let delay_byte = src.read_u8()?;
+    let delay_byte = raw_config_slice.read_u8()?;
     let initial_presentation_delay_present = (delay_byte & 0x10) == 0x10;
     let initial_presentation_delay_minus_one = if initial_presentation_delay_present {
         delay_byte & 0x0f
     } else {
         0
     };
 
-    let config_obus_size = src.bytes_left();
-    let config_obus = read_buf(src, config_obus_size)?;
+    let config_obus_size = raw_config_slice.len().to_u64();
+    let config_obus = read_buf(&mut raw_config_slice, config_obus_size)?;
```

Comment:
Alternatively, could we get rid of `config_obus` altogether? It doesn't appear to be used anywhere and this would save some unnecessary copies.

If we're concerned about the rust API providing access, I'd suggest we add
```rust
impl AV1ConfigBox {
    /// See https://aomediacodec.github.io/av1-isobmff/#av1codecconfigurationbox-syntax
    const CONFIG_OBUS_OFFSET: usize = 4;

    pub fn config_obus(&self) -> &[u8] {
        &self.raw_config[Self::CONFIG_OBUS_OFFSET..]
    }
}
```

Reply:
Thanks for the suggestions! I'll update the PR to use the second approach and expose the OBUs via a method to ensure the Rust API still exposes them.